File: src/Python/GeometricObjects/Polygon.py
#!/usr/bin/env python

import vtk

def main():
    colors = vtk.vtkNamedColors()

    # Setup four points
    points = vtk.vtkPoints()
    points.InsertNextPoint(0.0, 0.0, 0.0)
    points.InsertNextPoint(1.0, 0.0, 0.0)
    points.InsertNextPoint(1.0, 1.0, 0.0)
    points.InsertNextPoint(0.0, 1.0, 0.0)

    # Create the polygon
    polygon = vtk.vtkPolygon()
    polygon.GetPointIds().SetNumberOfIds(4)  # make a quad
    polygon.GetPointIds().SetId(0, 0)
    polygon.GetPointIds().SetId(1, 1)
    polygon.GetPointIds().SetId(2, 2)
    polygon.GetPointIds().SetId(3, 3)

    # Add the polygon to a list of polygons
    polygons = vtk.vtkCellArray()
    polygons.InsertNextCell(polygon)

    # Create a PolyData
    polygonPolyData = vtk.vtkPolyData()
    polygonPolyData.SetPoints(points)
    polygonPolyData.SetPolys(polygons)

    # Create a mapper and actor
    mapper = vtk.vtkPolyDataMapper()
    if vtk.VTK_MAJOR_VERSION <= 5:
        mapper.SetInput(polygonPolyData)
    else:
        mapper.SetInputData(polygonPolyData)

    actor = vtk.vtkActor()
    actor.SetMapper(mapper)
    actor.GetProperty().SetColor(colors.GetColor3d("Silver"))

    # Visualize
    renderer = vtk.vtkRenderer()
    renderWindow = vtk.vtkRenderWindow()
    renderWindow.SetWindowName("Polygon")
    renderWindow.AddRenderer(renderer)
    renderWindowInteractor = vtk.vtkRenderWindowInteractor()
    renderWindowInteractor.SetRenderWindow(renderWindow)

    renderer.AddActor(actor)
    renderer.SetBackground(colors.GetColor3d("Salmon"))
    renderWindow.Render()
    renderWindowInteractor.Start()


if __name__ == '__main__':
    main()

File: src/Java/GeometricObjects/Polygon.java
import vtk.vtkActor;
import vtk.vtkCellArray;
import vtk.vtkNamedColors;
import vtk.vtkNativeLibrary;
import vtk.vtkPoints;
import vtk.vtkPolyData;
import vtk.vtkPolyDataMapper;
import vtk.vtkPolygon;
import vtk.vtkRenderWindow;
import vtk.vtkRenderWindowInteractor;
import vtk.vtkRenderer;

public class Polygon  
{
	  
	// -----------------------------------------------------------------
	// Load VTK library and print which library was not properly loaded
	static 
  {
	  if (!vtkNativeLibrary.LoadAllNativeLibraries()) 
    {
	    for (vtkNativeLibrary lib : vtkNativeLibrary.values()) 
      {
	      if (!lib.IsLoaded()) 
        {
	        System.out.println(lib.GetLibraryName() + " not loaded");
	      }
	    }
	  }
	  vtkNativeLibrary.DisableOutputWindow(null);
	}
	// -----------------------------------------------------------------
	  

  public static void main(String s[]) 
  {

    vtkNamedColors colors = new vtkNamedColors();

    //For Actor Color
    double actorColor[] = new double[4];
    //Renderer Background Color
    double Bgcolor[] = new double[4];

    colors.GetColor("Red", actorColor);
    colors.GetColor("White", Bgcolor);

    vtkPoints Points = new vtkPoints();
    Points.InsertNextPoint(0.0,0.0,0.0);
    Points.InsertNextPoint(1.0,0.0,0.0);
    Points.InsertNextPoint(1.0,1.0,0.0);
    Points.InsertNextPoint(0.0,1.0,0.0);

          
    vtkPolygon Polygon = new vtkPolygon();
    Polygon.GetPointIds().SetNumberOfIds(4);
    Polygon.GetPointIds().SetId(0, 0);
    Polygon.GetPointIds().SetId(1, 1);
    Polygon.GetPointIds().SetId(2, 2);
    Polygon.GetPointIds().SetId(3, 3);

        
    vtkCellArray Polygons = new vtkCellArray();
    Polygons.InsertNextCell(Polygon);


    vtkPolyData PolygonPolyData = new vtkPolyData();
    PolygonPolyData.SetPoints(Points);
    PolygonPolyData.SetPolys(Polygons);


    //Create a Mapper and Actor
    vtkPolyDataMapper Mapper = new vtkPolyDataMapper();
    Mapper.SetInputData(PolygonPolyData);


    vtkActor Actor = new vtkActor();
    Actor.SetMapper(Mapper);
    Actor.GetProperty().SetColor(actorColor); 
    Actor.GetProperty().SetLineWidth(5);
        
    //Create the renderer, render window and interactor.
    vtkRenderer ren = new vtkRenderer();
    vtkRenderWindow renWin = new vtkRenderWindow();
    renWin.AddRenderer(ren);
    vtkRenderWindowInteractor iren = new vtkRenderWindowInteractor();
    iren.SetRenderWindow(renWin);

    // Visualise the arrow
    ren.AddActor(Actor);
    ren.SetBackground(Bgcolor);

    renWin.SetSize(300, 300);
    renWin.Render();

    iren.Initialize();
    iren.Start();	  
       
  }
} 
